code/modules/mob/living/carbon/xenomorph/hive_datum.dm

```diff
@@ -767,27 +773,42 @@
 
 // This proc attempts to find a new ruler to lead the hive.
 /datum/hive_status/proc/update_ruler()
-	if(living_xeno_ruler)
+	if(isxenoqueen(living_xeno_ruler))
 		return //No succession required.
 
 	var/mob/living/carbon/xenomorph/successor
+	var/list/mob/living/carbon/xenomorph/prio_candidates = xenos_by_tier[XENO_TIER_FOUR]
+	var/list/mob/living/carbon/xenomorph/seco_candidates = xenos_by_tier[XENO_TIER_THREE]
 
-	// TO DO: this shouldn't be a strict type check and should account for strains
-	var/list/candidates = xenos_by_typepath[/datum/xeno_caste/queen] + xenos_by_typepath[/datum/xeno_caste/shrike] + xenos_by_typepath[/datum/xeno_caste/king] + xenos_by_typepath[/datum/xeno_caste/dragon] + xenos_by_typepath[/datum/xeno_caste/king/conqueror]
-	if(length(candidates)) //Priority to the queens.
-		successor = candidates[1] //First come, first serve.
+	if(prio_candidates.len > 0)
+		for(var/mob/living/carbon/xenomorph/potential_successor in prio_candidates)
+			if(isxenoqueen(potential_successor))
+				successor = potential_successor
+				break
+			else
+				successor = potential_successor
+	else if(seco_candidates.len > 0)
+		for(var/mob/living/carbon/xenomorph/potential_successor in seco_candidates)
+			if(potential_successor.xeno_caste.can_flags & CASTE_CAN_BE_RULER && !living_xeno_ruler)
+				successor = potential_successor
+			else
+				return
+
```

Comment:
return?what?

Reply:
This entire thing gets called everytime anyone evolves to anything. So we dont wanna run the rest of the code if nobody in tier 3 or 4 exists.

Reply:
Removed